components/script/dom/htmlimageelement.rs

```diff
@@ -782,6 +798,63 @@ impl LayoutHTMLImageElementHelpers for LayoutJS<HTMLImageElement> {
     }
 }
 
+pub fn parse_a_sizes_attribute(input: DOMString, width: Option<u32>) -> Vec<Size> {
+    let mut sizes = Vec::<Size>::new();
+    let unparsed_sizes = input.split(',').collect::<Vec<_>>();
+
+    for unparsed_size in &unparsed_sizes {
+        let whitespace = unparsed_size.chars().rev().take_while(|c| char::is_whitespace(*c)).count();
+        let trimmed: String = unparsed_size.chars().take(unparsed_size.chars().count() - whitespace).collect();
+
+        if trimmed.is_empty() {
+            continue;
+        }
+        let mut input = ParserInput::new(&trimmed);
+        let url = ServoUrl::parse("about:blank").unwrap();
+        let reporter = NullReporter;
+        let context = ParserContext::new_for_cssom(&url,
+                                                   &reporter,
+                                                   None,
+                                                   ParsingMode::empty(),
+                                                   QuirksMode::NoQuirks);
+        let length = Parser::new(&mut input).try(|i| Length::parse_non_negative(&context, i));
```

Comment:
We should create a separate parser object that we can reuse later.

Reply:
How to create a separate parser object here? I am not clear if we can create a Parser object without any input string and reuse it later.

Reply:
```rust
let mut parser = Parser::new(&mut input);
/* ... */
let length = parser.try(..);
/* .. */
let something_else = parser.foo(..).
```